torch/testing/_internal/common_methods_invocations.py

```diff
@@ -4,12 +4,122 @@
 from operator import mul, itemgetter
 import collections
 from torch.autograd import Variable
-from torch.testing import make_non_contiguous
-from torch.testing._internal.common_device_type import skipCUDAIfNoMagma, skipCPUIfNoLapack, expectedFailureCUDA
-from torch.testing._internal.common_utils import (prod_single_zero, random_square_matrix_of_rank,
-                                                  random_symmetric_matrix, random_symmetric_psd_matrix,
-                                                  random_symmetric_pd_matrix, make_nonzero_det,
-                                                  random_fullrank_matrix_distinct_singular_value, set_rng_seed)
+from torch.testing import \
+    (make_non_contiguous,
+     get_common_float_types, get_all_float_types,
+     get_common_float_and_complex_types_plus_half,
+     get_common_float_and_complex_types_plus_bfloat16,
+     get_all_float_and_complex_types)
+from torch.testing._internal.common_device_type import \
+    (skipCUDAIfNoMagma, skipCPUIfNoLapack, expectedFailureCUDA)
+from torch.testing._internal.common_utils import \
+    (prod_single_zero, random_square_matrix_of_rank,
+     random_symmetric_matrix, random_symmetric_psd_matrix,
+     random_symmetric_pd_matrix, make_nonzero_det,
+     random_fullrank_matrix_distinct_singular_value, set_rng_seed)
+
+
+# Classes and methods for the operator database
+# The OpMeta class contains metadata and helpers for acquiring properties of
+# operations, like what dtypes they support.
+class OpMeta(object):
```

Comment:
We live in the future; consider making this a `@dataclass`

Reply:
Also flagging a potential naming collision: we already have a concept of "meta function" for an operator in PyTorch. The meta function computes all of the shape/dtype properties of the output tensor, but not the actual computation itself.

Reply:
I renamed this class to OpInfo.